frontend/src/components/general/form/FormForTreecluster.tsx

```diff
@@ -0,0 +1,98 @@
+import React from "react";
+import PrimaryButton from "../buttons/PrimaryButton";
+import Input from "./types/Input";
+import Select from "./types/Select";
+import Textarea from "./types/Textarea";
+import SelectTrees from "./types/SelectTrees";
+import { FieldErrors, SubmitHandler, UseFormRegister } from "react-hook-form";
+import { SoilConditionOptions } from "@/hooks/useDetailsForSoilCondition";
+import { TreeclusterSchema } from "@/schema/treeclusterSchema";
+import useFormStore, { FormStore } from "@/store/form/useFormStore";
+
+interface FormForTreeclusterProps {
+  displayError: boolean;
+  register: UseFormRegister<TreeclusterSchema>;
+  errors: FieldErrors<TreeclusterSchema>;
+  onSubmit: SubmitHandler<TreeclusterSchema>;
+  handleSubmit: (
+    onSubmit: SubmitHandler<TreeclusterSchema>,
+  ) => (e?: React.BaseSyntheticEvent) => Promise<void>;
+  onAddTrees: () => void;
+}
+
+const FormForTreecluster: React.FC<FormForTreeclusterProps> = ({
+  handleSubmit,
+  displayError,
+  register,
+  errors,
+  onSubmit,
+  onAddTrees,
+}) => {
+  const { form, set } = useFormStore((state: FormStore<TreeclusterSchema>) => ({
+    form: state.form,
+    set: state.commit,
+  }));
+
+  const handleDeleteTree = (treeId: number) => {
+    form &&
+      set({
+        ...form,
+        treeIds: form?.treeIds.filter((id) => id !== treeId) ?? [],
+      });
+  };
+
+  return (
+    <form
+      key="cluster-register"
+      className="space-y-6 lg:space-y-0 lg:grid lg:grid-cols-2 lg:gap-11"
+      onSubmit={handleSubmit(onSubmit)}
+    >
+      <div className="space-y-6">
+        <Input
+          label="Name"
+          error={errors.name?.message}
+          required
+          {...register("name", {})}
+        />
+        <Input
```

Comment:
Then we need to split the view with different zod schemas. I have built this so that everything depends on the zod schema, so if create and update are different, we need to define the zod schemas and build specific page with these input fields